### PR TITLE
Adding SpinAppCRD documentation

### PR DESCRIPTION
**Pull request by tpmccallum**
Signed-off-by: tpmccallum tim.mccallum@fermyon.com

Closes https://github.com/spinkube/documentation/issues/73

### CODE REVIEW

File: content/en/docs/glossary/_index.md [hunk 1]
```diff
@@ -32,7 +32,12 @@ In the context of Kubernetes, a Custom Resource (CR) is an extension mechanism t
 
 A Custom Resource Definition (CRD) is an extension mechanism that allows users to define their own custom resources. It enables the creation of new resource types with specific schemas and behaviors. CRDs define the structure and validation rules for custom resources, allowing users to store and manage additional information beyond the built-in Kubernetes resources. Once a CRD is created, instances of the custom resource can be created, updated, and deleted using the Kubernetes API. CRDs provide a way to extend Kubernetes and tailor it to specific application requirements.
 
-For example, the following `.yaml` file describes a `SpinApp` using CRD syntax:
+## SpinAppCRD
+
+The SpinApp CRD is a Kubernetes resource that extends the functionality of the Kubernetes API to support Spin applications. It defines a custom resource called "SpinApp" that encapsulates all the necessary information to deploy and manage a Spin application within a Kubernetes cluster. The SpinAppCRD consists of several key fields that define the desired state of a Spin application. 
+
```
Comment (tpmccallum) [inline content/en/docs/glossary/_index.md:38]:
```suggestion
```

File: content/en/docs/glossary/_index.md [hunk 2]
```diff
@@ -45,7 +49,56 @@ spec:
...
+    - name: ENV_VAR2
+      value: value2
+  # Add any other user-defined values here
+```
+
+In this updated example, we have added additional customization options:
+
+- `imagePullSecrets`: An optional field that lets you reference a Kubernetes secret that has credentials for you to pull in image from a private registry.
+- `serviceAnnotations`: An optional field that lets you set specific annotations on the underlying service that is created.
+- `deploymentAnnotations`: An optional field that lets you set specific annotations on the underlying deployment that is created.
```
Comment (tpmccallum) [inline content/en/docs/glossary/_index.md:92]:
```suggestion
```

File: content/en/docs/glossary/_index.md [hunk 3]
```diff
@@ -45,7 +49,56 @@ spec:
...
+metadata:
+  name: simple-spinapp
+spec:
+  image: 'ghcr.io/deislabs/containerd-wasm-shims/examples/spin-rust-hello:v0.10.0'
+  replicas: 3
+  imagePullSecrets:
+    - name: spin-image-secret
+  serviceAnnotations:
+    key: value
+  deploymentAnnotations:
```
Comment (tpmccallum) [inline content/en/docs/glossary/_index.md:68]:
```suggestion
```

File: content/en/docs/glossary/_index.md [hunk 4]
```diff
@@ -45,7 +49,56 @@ spec:
...
+    key: value
```
Comment (tpmccallum) [inline content/en/docs/glossary/_index.md:69]:
```suggestion
```

File: content/en/docs/glossary/_index.md [hunk 5]
```diff
@@ -45,7 +49,56 @@ spec:
...
+    multiple-keys: are-supported
```
Comment (tpmccallum) [inline content/en/docs/glossary/_index.md:70]:
```suggestion
```

**Comment by tpmccallum** [comment]
PR here https://github.com/spinkube/documentation/pull/95

File: content/en/docs/glossary/_index.md [hunk 6]
```diff
@@ -32,7 +32,11 @@ In the context of Kubernetes, a Custom Resource (CR) is an extension mechanism t
 
 A Custom Resource Definition (CRD) is an extension mechanism that allows users to define their own custom resources. It enables the creation of new resource types with specific schemas and behaviors. CRDs define the structure and validation rules for custom resources, allowing users to store and manage additional information beyond the built-in Kubernetes resources. Once a CRD is created, instances of the custom resource can be created, updated, and deleted using the Kubernetes API. CRDs provide a way to extend Kubernetes and tailor it to specific application requirements.
 
-For example, the following `.yaml` file describes a `SpinApp` using CRD syntax:
+## SpinAppCRD
+
+The SpinApp CRD is a Kubernetes resource that extends the functionality of the Kubernetes API to support Spin applications. It defines a custom resource called "SpinApp" that encapsulates all the necessary information to deploy and manage a Spin application within a Kubernetes cluster. The SpinAppCRD consists of several key fields that define the desired state of a Spin application.
```
Comment (vdice) [inline content/en/docs/glossary/_index.md:37]:
```suggestion
The SpinApp CRD is a Kubernetes resource that extends the functionality of the Kubernetes API to support Spin applications. It defines a custom resource called "SpinApp" that encapsulates all the necessary information to deploy and manage a Spin application within a Kubernetes cluster. The SpinApp CRD consists of several key fields that define the desired state of a Spin application. 
```

File: content/en/docs/glossary/_index.md [hunk 7]
```diff
@@ -32,7 +32,11 @@ In the context of Kubernetes, a Custom Resource (CR) is an extension mechanism t
 
 A Custom Resource Definition (CRD) is an extension mechanism that allows users to define their own custom resources. It enables the creation of new resource types with specific schemas and behaviors. CRDs define the structure and validation rules for custom resources, allowing users to store and manage additional information beyond the built-in Kubernetes resources. Once a CRD is created, instances of the custom resource can be created, updated, and deleted using the Kubernetes API. CRDs provide a way to extend Kubernetes and tailor it to specific application requirements.
 
-For example, the following `.yaml` file describes a `SpinApp` using CRD syntax:
+## SpinAppCRD
```
Comment (vdice) [inline content/en/docs/glossary/_index.md:35]:
```suggestion
## SpinApp CRD
```

File: content/en/docs/glossary/_index.md [hunk 8]
```diff
@@ -45,7 +49,52 @@ spec:
   runtime: "containerd-shim-spin"
 ```
 
-> SpinApp CRDs are kept separate from Helm. If using Helm, CustomResourceDefinition (CRD) resources will need to be installed prior to installing the Heml chart.
+> SpinApp CRDs are kept separate from Helm. If using Helm, CustomResourceDefinition (CRD) resources must be installed prior to installing the Heml chart.
```
Comment (vdice) [inline content/en/docs/glossary/_index.md:52]:
```suggestion
> SpinApp CRDs are kept separate from Helm. If using Helm, CustomResourceDefinition (CRD) resources must be installed prior to installing the Helm chart.
```

File: content/en/docs/glossary/_index.md [hunk 9]
```diff
@@ -45,7 +49,52 @@ spec:
...
+
+You can modify the example above to customize the SpinAppCRD via a YAML file. Here's an updated YAML file with additional customization options:
```
Comment (vdice) [inline content/en/docs/glossary/_index.md:54]:
```suggestion
You can modify the example above to customize the SpinApp via a YAML file. Here's an updated YAML file with additional customization options:
```

File: content/en/docs/glossary/_index.md [hunk 10]
```diff
@@ -45,7 +49,52 @@ spec:
...
+    - name: ENV_VAR1
+      value: value1
+    - name: ENV_VAR2
+      value: value2
+  # Add any other user-defined values here
+```
+
+In this updated example, we have added additional customization options:
+
+- `imagePullSecrets`: An optional field that lets you reference a Kubernetes secret that has credentials for you to pull in image from a private registry.
```
Comment (vdice) [inline content/en/docs/glossary/_index.md:87]:
```suggestion
- `imagePullSecrets`: An optional field that lets you reference a Kubernetes secret that has credentials for you to pull in images from a private registry.
```

File: content/en/docs/glossary/_index.md [hunk 11]
```diff
@@ -45,7 +49,52 @@ spec:
...
+- `serviceAnnotations`: An optional field that lets you set specific annotations on the underlying service that is created.
+- `podAnnotations`: An optional field that lets you set specific annotations on the underlying pods that are created.
+- `resources`: You can specify resource limits and requests for CPU and memory. Adjust the values according to your application's resource requirements.
+- `env`: You can define environment variables for your SpinApp. Add as many environment variables as needed, providing the name and value for each.
+
+To apply the changes, save the YAML file (e.g. `updated-spinapp-crd.yaml`) and then apply it to your Kubernetes cluster using the following command:
```
Comment (vdice) [inline content/en/docs/glossary/_index.md:93]:
```suggestion
To apply the changes, save the YAML file (e.g. `updated-spinapp.yaml`) and then apply it to your Kubernetes cluster using the following command:
```

File: content/en/docs/glossary/_index.md [hunk 12]
```diff
@@ -45,7 +49,52 @@ spec:
...
+
+```bash
+kubectl apply -f updated-spinapp-crd.yaml
```
Comment (vdice) [inline content/en/docs/glossary/_index.md:96]:
```suggestion
kubectl apply -f updated-spinapp.yaml
```

File: content/en/docs/glossary/_index.md [hunk 13]
```diff
@@ -32,7 +32,11 @@ In the context of Kubernetes, a Custom Resource (CR) is an extension mechanism t
 
 A Custom Resource Definition (CRD) is an extension mechanism that allows users to define their own custom resources. It enables the creation of new resource types with specific schemas and behaviors. CRDs define the structure and validation rules for custom resources, allowing users to store and manage additional information beyond the built-in Kubernetes resources. Once a CRD is created, instances of the custom resource can be created, updated, and deleted using the Kubernetes API. CRDs provide a way to extend Kubernetes and tailor it to specific application requirements.
 
-For example, the following `.yaml` file describes a `SpinApp` using CRD syntax:
+## SpinAppCRD
+
+The SpinApp CRD is a Kubernetes resource that extends the functionality of the Kubernetes API to support Spin applications. It defines a custom resource called "SpinApp" that encapsulates all the necessary information to deploy and manage a Spin application within a Kubernetes cluster. The SpinAppCRD consists of several key fields that define the desired state of a Spin application. 
+
+Here's an example of a SpinAppCRD definition:
```
Comment (vdice) [inline content/en/docs/glossary/_index.md:39]: Pedantic update - difference between CRD and custom resources (or objects) that are based on that CRD.  Perhaps something like the following?

```suggestion
Here's an example of a SpinApp custom resource that uses the SpinApp CRD schema:
```

**Review by vdice — APPROVED** [review]
Mainly just notes around distinction between CRD and custom objects derived from it (in this case for a SpinApp).

Otherwise LGTM!

**Pull request merged**